sql/sqlliveness/slstorage: TestDeleteMidUpdateFails failed

sql/sqlliveness/slstorage.TestDeleteMidUpdateFails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14747972?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14747972?buildTab=artifacts#/) on master @ [3ba9c4873d4422ea1ff640e47c5fa9954ebbb3a6](https://github.com/cockroachdb/cockroach/commits/3ba9c4873d4422ea1ff640e47c5fa9954ebbb3a6):

Fatal error:

```
panic: test timed out after 57s
running tests:
	TestDeleteMidUpdateFails (10s)

```
Stack: 

```
goroutine 19269 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestDeleteMidUpdateFails
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/7ac62741d99709d7571b3ca7444cf377/logTestDeleteMidUpdateFails774625519
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    slstorage_test.go:736: -- test log scope end --
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDeleteMidUpdateFails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37644

Seems like the tests in this package might be a little too expensive for the "small" designation:
```
--- PASS: TestStorage (7.62s)
    --- PASS: TestStorage/basic-insert-is-alive (0.05s)
    --- PASS: TestStorage/delete-update (0.04s)
    --- PASS: TestStorage/delete-expired-on-is-alive (0.04s)
    --- PASS: TestStorage/test-jitter (0.05s)
=== RUN   TestConcurrentAccessesAndEvictions
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/7ac62741d99709d7571b3ca7444cf377/logTestConcurrentAccessesAndEvictions1036881751
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    slstorage_test.go:453: -- test log scope end --
--- PASS: TestConcurrentAccessesAndEvictions (17.09s)
=== RUN   TestConcurrentAccessSynchronization
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/7ac62741d99709d7571b3ca7444cf377/logTestConcurrentAccessSynchronization712242487
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestConcurrentAccessSynchronization/CachedReader_does_not_block
=== RUN   TestConcurrentAccessSynchronization/canceled_context_does_not_interrupt_read
=== RUN   TestConcurrentAccessSynchronization/context_cancellation_returns
=== NAME  TestConcurrentAccessSynchronization
    slstorage_test.go:648: -- test log scope end --
--- PASS: TestConcurrentAccessSynchronization (22.35s)
    --- PASS: TestConcurrentAccessSynchronization/CachedReader_does_not_block (0.00s)
    --- PASS: TestConcurrentAccessSynchronization/canceled_context_does_not_interrupt_read (0.00s)
    --- PASS: TestConcurrentAccessSynchronization/context_cancellation_returns (0.00s)
=== RUN   TestDeleteMidUpdateFails
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/7ac62741d99709d7571b3ca7444cf377/logTestDeleteMidUpdateFails774625519
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    slstorage_test.go:736: -- test log scope end --
panic: test timed out after 57s
running tests:
	TestDeleteMidUpdateFails (10s)

goroutine 19269 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```

--- a/pkg/sql/row/BUILD.bazel
+++ b/pkg/sql/row/BUILD.bazel
@@ -94,7 +94,7 @@ go_library(
 
 go_test(
     name = "row_test",
-    size = "small",
+    size = "medium",
     srcs = [
         "expr_walker_test.go",
         "fetcher_mvcc_test.go",

--- a/pkg/sql/sqlliveness/slstorage/BUILD.bazel
+++ b/pkg/sql/sqlliveness/slstorage/BUILD.bazel
@@ -44,7 +44,7 @@ go_library(
 
 go_test(
     name = "slstorage_test",
-    size = "small",
+    size = "medium",
     srcs = [
         "key_encoder_test.go",
         "main_test.go",

